src/main/java/task/Task.java

```diff
@@ -0,0 +1,179 @@
+package task;
+
+import duke.Duke;
+import java.time.LocalDateTime;
+
+
+public class Task {
+
+    private String taskName;
+    private Boolean isDone;
+    private LocalDateTime dateAdded;
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName) {
+        return new ToDos(taskName);
+    }
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @param dayDate The date and time of the deadline of the task.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName, LocalDateTime dayDate) {
+        return new Deadlines(taskName, dayDate);
+    }
+
+    /**
+     * Calls the constructor to create a Task.
+     *
+     * @param taskName The name of the task.
+     * @param startDayDateTime The date and time of the start of the event.
+     * @param endDayDateTime The date and time of the end of the event.
+     * @return Returns a Task object.
+     */
+    public static Task of(String taskName, LocalDateTime startDayDateTime, LocalDateTime endDayDateTime) {
+        return new Event(taskName, startDayDateTime, endDayDateTime);
+    }

```

Comment:
These constructors seem a little redundant, you could call the child class constructors directly after putting each child class into it's own file.